packages/thirdweb/src/event/actions/get-events.ts

```diff
@@ -186,3 +229,98 @@ export async function getContractEvents<
     events: resolvedEvents,
   });
 }
+
+async function getLogsFromInsight(options: {
+  params: GetLogsParamsExtra;
+  chain: Chain;
+  client: ThirdwebClient;
+  signature?: string;
+}): Promise<Log[]> {
+  const { params, chain, client, signature } = options;
+
+  const chainServices = await getChainServices(chain);
+  const insightEnabled = chainServices.some(
+    (c) => c.service === "insight" && c.enabled,
+  );
+
+  if (!insightEnabled) {
+    throw new Error(`Insight is not available for chainId ${chain.id}`);
+  }
+
+  try {
+    let baseUrl = `https://${getThirdwebDomains().insight}/v1/events`;
+    if (params.address) {
+      baseUrl += `/${params.address}`;
+      if (signature) {
+        baseUrl += `/${signature}`;
+      }
+    }
+    const url = new URL(baseUrl);
+
+    url.searchParams.set("chain", chain.id.toString());
+    url.searchParams.set("limit", "500"); // this is max limit on insight
+
+    if (params.blockHash) {
+      url.searchParams.set("filter_block_hash", params.blockHash);
+    } else {
+      if (params.fromBlock) {
+        const fromBlock =
+          typeof params.fromBlock === "bigint"
+            ? numberToHex(params.fromBlock)
+            : params.fromBlock;
+
+        url.searchParams.set("filter_block_number_gte", fromBlock);
+      }
+      if (params.toBlock) {
+        const toBlock =
+          typeof params.toBlock === "bigint"
+            ? numberToHex(params.toBlock)
+            : params.toBlock;
+
+        url.searchParams.set("filter_block_number_lte", toBlock);
+      }
+    }
+
+    if (params.topics) {
+      const args = params.topics.slice(1);
+      for (const [i, a] of args.entries()) {
+        if (a) {
+          url.searchParams.set(`filter_topic_${i + 1}`, a as Hex);
+        }
+      }
+    }
+
+    const clientFetch = getClientFetch(client);
+    const result = await clientFetch(url.toString());
```

Comment:
should always check for error on response btw. and parse the error. will fix it when i switch to the autogen api though